softlayer/resource_softlayer_global_ip_test.go

```diff
@@ -0,0 +1,155 @@
+package softlayer
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+	"github.com/softlayer/softlayer-go/services"
+	"github.com/softlayer/softlayer-go/session"
+)
+
+func TestAccSoftLayerGlobalIp_Basic(t *testing.T) {
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccCheckSoftLayerGlobalIpConfig_basic,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSoftLayerGlobalIpExists("softlayer_global_ip.test-global-ip"),
+					testAccCheckSoftLayerResources("softlayer_global_ip.test-global-ip", "routes_to",
+						"softlayer_virtual_guest.vm1", "ipv4_address"),
+				),
```

Comment:
I would add another check to ensure ip_address in the global ip resource is set using `resource.TestMatchResourceAttr()`